ode-wrapper/src/main/java/com/trihydro/odewrapper/helpers/SetItisCodes.java

```diff
@@ -157,30 +160,72 @@ public List<String> setItisCodes(WydotTim wydotTim) {
                 continue;
             }
 
-            code = getItisCodes().stream().filter(x -> x.getItisCode().equals(itisCode)).findFirst().orElse(null);
+            code = getAllItisCodesFromDatabase().stream().filter(x -> x.getItisCode().equals(itisCode)).findFirst().orElse(null);
 
             if (code != null)
                 items.add(code.getItisCode().toString());
         }
 
-    return items;
-  }
+        return items;
+    }
+
+    /**
+     * Creates a properly ordered list of ITIS codes based on the TIM.
+     *
+     * @param wydotTim TIM to base the ITIS codes on
+     * @return Properly ordered list of ITIS codes to assign back to the TIM
+     */
+    public List<String> setItisCodesVsl(WydotTimVsl wydotTim) {
+
+        List<String> items = new ArrayList<String>();
+        if (wydotTim.getSpeed() == null) {
+            throw new IllegalArgumentException("Cannot determine Speed limit. Speed cannot be null.");
+        }
+
+        List<ItisCode> allItisCodes = getAllItisCodesFromDatabase();
+
+        // add advisory code first if advisory speed limit
+        wydotTim.getItisCodes().stream().filter(x -> x.contains(ADVISORY_ITIS_CODE)).findFirst()
+                .ifPresent(items::add);
 
-  public List<String> setItisCodesBowr(WydotTimBowr tim) throws WeightNotSupportedException {
-    List<String> itisCodes = new ArrayList<>();
+        // speed limit itis code
+        allItisCodes.stream().filter(x -> x.getDescription().equals("speed limit")).findFirst().ifPresent(mph -> items.add(mph.getItisCode().toString()));
 
-    int weightInPounds = tim.getData();
+        // add reduced code next if reduced speed
+        wydotTim.getItisCodes().stream().filter(x -> x.contains(REDUCED_ITIS_CODE)).findFirst()
+                .ifPresent(items::add);
 
-    itisCodes.add("5127"); // Strong winds
-    itisCodes.add("2563"); // Truck restriction
-    itisCodes.add("2569"); // No high profile vehicles
-    itisCodes.add("7682"); // Below
-    itisCodes.add("2577"); // Gross-Weight-Limit
-    itisCodes.add(translateWeightToItisCode(weightInPounds)); // Weight, translated from pounds to ITIS code
-    itisCodes.add("8739"); // Pounds
+        // J2540 small number ITIS codes start at 12,544 and increment by 1 to represent each number up to 255.
+        // To covert speed to an ITIS code, adding 12,544 is enough.
+        items.add(Integer.toString(
+                Optional.ofNullable(wydotTim.getSpeed()).orElse(0) + 12544
+        ));

```

Comment:
suggestion(non-blocking): to improve clarity of intent ease readability, we can do the following.

```suggestion
        items.add(Integer.toString(wydotTim.getSpeed() + SMALL_ITIS_CODE_MIN));
```

with:
```java
private static final int SMALL_ITIS_CODE_MIN = 12544;
```

On line 182, we already throw an exception if speed is null, so the optional.ofNullable isn't necessary (it will never be null). Adding a named constant for the minimum small itis code self-documents the intent of that number and makes future refactors (if we need that number elsewhere) a little easier.

Reply:
Should be applied now, but since I resolved it on the branch itself, it doesn't show it as committed.